timemachine/cpp/src/nonbonded_interaction_group.cu

```diff
@@ -393,12 +361,16 @@ void NonbondedInteractionGroup<RealType>::du_dp_fixed_to_float(
     const int N, const int P, const unsigned long long *du_dp, double *du_dp_float) {
 
     for (int i = 0; i < N; i++) {
-        const int idx_charge = i * 3 + 0;
-        const int idx_sig = i * 3 + 1;
-        const int idx_eps = i * 3 + 2;
+        const int idx = i * PARAMS_PER_ATOM;
+        const int idx_charge = idx + PARAM_OFFSET_CHARGE;
+        const int idx_sig = idx + PARAM_OFFSET_SIG;
+        const int idx_eps = idx + PARAM_OFFSET_EPS;
+        const int idx_w = idx + PARAM_OFFSET_W;
+
         du_dp_float[idx_charge] = FIXED_TO_FLOAT_DU_DP<double, FIXED_EXPONENT_DU_DCHARGE>(du_dp[idx_charge]);
         du_dp_float[idx_sig] = FIXED_TO_FLOAT_DU_DP<double, FIXED_EXPONENT_DU_DSIG>(du_dp[idx_sig]);
         du_dp_float[idx_eps] = FIXED_TO_FLOAT_DU_DP<double, FIXED_EXPONENT_DU_DEPS>(du_dp[idx_eps]);
+        du_dp_float[idx_w] = FIXED_TO_FLOAT<double>(du_dp[idx_w]);
```

Comment:
Question: Should the fixed exponent for `du_dw` be exposed separately? (Or is it okay to keep it tied to global `FIXED_EXPONENT`?)

Reply:
I was on the fence about this; was thinking to do this later if we find it necessary in some case, but for now it seemed simplest to use the same fixed exponent that we use for the other positional gradients.

Reply:
Makes sense!

(To clarify, I wasn't suggesting to pick a different value than `FIXED_EXPONENT` initially, just to add a `static const auto FIXED_EXPONENT_DU_DW = FIXED_EXPONENT;` or something to https://github.com/proteneer/timemachine/blob/master/timemachine/cpp/src/fixed_point.hpp , and add references to it from lines like this, so that (1) it's initially the same as `FIXED_EXPONENT`, (2) if tuning it separately becomes needed, only one site will need to be updated. It does seem quite different to me from spatial gradients, so I imagine the need could arise to tune a separate exponent here.)

Reply:
Good call, I think this makes sense. Added in 08ddb1d. This also led me to discover a bug where I'd used the charge exponent by mistake (which just happens to be the same as the one for dU/dx)